src/init.cpp

```diff
@@ -1043,9 +1065,46 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
                 delete pcoinsdbview;
                 delete pcoinscatcher;
                 delete pblocktree;
+                delete pblocktreeother;
+                delete pblockdb;
+                delete pblockundodb;
 
                 uiInterface.InitMessage(_("Opening Block database..."));
-                pblocktree = new CBlockTreeDB(nBlockTreeDBCache, false, fReindex);
+                if (BLOCK_DB_MODE == DB_BLOCK_STORAGE)
```

Comment:
Let's add an "InitializeBlockStorage" API to wrapper.h, and impl in the wrapper.c.  In DB_BLOCK_STORAGE mode it does all that's below.  In SEQUENTIAL mode it does the code from lines 1005 to approx 1037.  This isolates and abstracts the block storage initialization logic.

Reply:
the block file format that the upgrade code on lines 1005 to 1037 addresses havent been used in years, isn't it time to remove this all together?